src/main/fc/rc_adjustments.c

```diff
@@ -221,7 +223,29 @@ static const adjustmentConfig_t defaultAdjustmentConfigs[ADJUSTMENT_FUNCTION_COU
         .adjustmentFunction = ADJUSTMENT_HORIZON_STRENGTH,
         .mode = ADJUSTMENT_MODE_SELECT,
         .data = { .switchPositions = 255 }
+    },
+#ifdef USE_OSD_PROFILES
+    // the entries below are required because the entry for ADJUSTMENT_OSD_PROFILE below must
+    // correspond with its position in the adjustmentFunction_e enum.
+    {
```

Comment:
I think #6796 should have eliminated the need for these filler elements - can you please check again?

Reply:
@mikeller No worries.  I have checked #6796, it does not apply to this array, i.e. it is still indexed in the same way.  I have followed ledvinap's suggestion above and am now using the code below, hence the dummy entries are no longer required - I think this is better.
[ADJUSTMENT_OSD_PROFILE - ADJUSTMENT_FUNCTION_CONFIG_INDEX_OFFSET] =
    {
        .adjustmentFunction = ADJUSTMENT_OSD_PROFILE,
        .mode = ADJUSTMENT_MODE_SELECT,
        .data = { .switchPositions = 3 }
    }

Reply:
@mikeller Another question.  I have noticed in my testing that get osd_profile index in the CLI returns:

# get osd_profile
osd_profile = 1
Allowed range: 1 - 3
Default value: 0 ###CORRUPTED CONFIG###

I want to make the default 1, but cannot find where to specify this.  For most osd elements it is 234, but I cannot find how to specify this.  Can you or someone please point me in the right direction?  Thank you.

Reply:
@pkruger: Try applying #7068 to see if this fixes the problem. The reset to defaults is done in `pgResetFn_osdConfig()` in `src/main/io/osd.c`.

Reply:
@mikeller I'm confused, not sure how #7068 will make a difference, the array is still indexed in the same way, seems only the variable is renamed, the mechanism is still the same:
const adjustmentConfig_t *adjustmentConfig = &defaultAdjustmentConfigs[adjustmentRange->adjustmentConfig - ADJUSTMENT_FUNCTION_CONFIG_INDEX_OFFSET];

Thanks for the pointer to the defaults - fixed now.

Reply:
This is the change that is fixing the problem with the sparse array:

https://github.com/betaflight/betaflight/pull/7068/files#diff-9ef0580402865270af23722cc039390aL805

After this, everything should be indexing correctly.

Reply:
@etracer65: Do you have an idea why this still isn't working as it should? I think instead of padding the `defaultAdjustmentConfigs`, we should fix the underlying issue.

Reply:
@mikeller I think this PR just needs to be rebased. #7068 will fix any problems with `defaultAdjustmentConfigs` being aligned differently than `adjustmentFunction_e`. In fact adding padding or otherwise changing the order of `defaultAdjustmentConfigs` will actually introduce bugs in that it's no longer aligned with the Configurator and this will corrupt the config.  The key misconception here is that `defaultAdjustmentConfigs` should be aligned with `adjustmentFunction_e` - this is not true. `defaultAdjustmentConfigs` **MUST** be aligned with the Configurator. Each entry in `defaultAdjustmentConfigs` references the appropriate `adjustmentFunction_e` index that it relates to.

@pkruger 
> So I tried 7068 mentioned above, unfortunately it still doesn't work because defaultAdjustmentConfigs is still indexed in the same way. The array index of defaultAdjustmentConfigs must still correspond with the position in the enum, adjustmentFunction_e.

This is not true. Besides the variable renaming to make it less confusing in #7068, The key fix is:

```
-             applyAbsoluteAdjustment(controlRateConfig, adjustmentRange->adjustmentFunction, value);
+             applyAbsoluteAdjustment(controlRateConfig, adjustmentConfig->adjustmentFunction, value);
```
The key point is that `adjustmentRange` references the index of `adjustmentConfig` and `adjustmentConfig` references the `adjustmentFunction_e` index.  So the base code is working correctly now. I haven't looked at this PR in detail but maybe you have code making the same mistake?

Reply:
Also I just tested adding an `ADJUSTMENT_MODE_SELECT` type entry to the end of the `defaultAdjustmentConfigs` array (actually duplicated the `ADJUSTMENT_RATE_PROFILE` entry so it was the 29th entry instead of the 12th and not related to `adjustmentFunction_e` ) and it worked as expected - no padding or other ordering changes needed.  Related CLI setting was: 
```
adjrange 0 0 2 900 2100 29 2 0 0
```

Reply:
Ok, I took this branch, rebased it to current master and removed the padding elements in `defaultAdjustmentConfigs`.  Also had to add `USE_OSD_PROFILES` to targets/common_pre.h as it was never defined.

Tested with settings `adjrange 0 0 2 900 2100 29 2 0 0` and profile changing worked as expected. The `osd_profile` parameter changed from 1, 2 or 3 depending on my switch setting. Note that I didn't test anything related to the OSD elements appearing based on the profile, just that the `osd_profile` value was correctly responding to the rc adjustment.

Perhaps you had the incorrect index in your `adjrange` setting? It should be 29 as noted above as this references the position in the `defaultAdjustmentConfigs` array. Maybe you used the index position for `ADJUSTMENT_OSD_PROFILE` from `adjustmentFunction_e` which would be incorrect and point an uninitialized entry in `defaultAdjustmentConfigs`.

Not also a necessary fix is that you must add:
```
#define USE_OSD_PROFILES
```
to the `#if (FLASH_SIZE > 256)` section of targets/common_pre.h (preferably in alphabetical order).

Reply:
@etracer65 : IMO it makes little sense to use (`static`) array index in configurator. Before this PR `defaultAdjustmentConfigs` was aligned with `adjustmentFunction_e` (with empty entries at end of the array). Breaking this will only make things complicated and harder to fix in future ...

Reply:
@ledvinap No, the "disconnect" of the alignment between `defaultAdjustmentConfigs` and `adjustmentFunction_e` happened well previous to this PR. The point is that there's a fundamental misunderstanding that the index used in the configurator refers to the `adjustmentFunction_e` index. This is not true. It refers to the index in `defaultAdjustmentConfigs` and each entry in there has a reference to which function (enum index) it's for.  `defaultAdjustmentConfigs` defines the available actions (thing to change and how to change it).  The enumeration just defines the "things" that can be changed - not how.  The point is that no padding is or was ever needed in `defaultAdjustmentConfigs` and it never had to align with `adjustmentFunction_e`. The code in this PR works correctly once the erroneous padding is removed.